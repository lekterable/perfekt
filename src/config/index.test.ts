import defaultConfig from './default'
import Config from '.'

describe('Config', () => {
  const { config } = new Config()

  describe('get config', () => {
    it('should return default config', () => {
      expect(config).toStrictEqual(defaultConfig)
      expect(config).toMatchSnapshot()
    })

    it('should return config with overrides', () => {
      const overrides = {
        unreleasedHeader: '## **Unreleased**',
        releaseHeader: '## Release version - @%version%',
        breakingHeader: '# BREAKING',
        lineFormat: '* %message% %hash% %hash%'
      }
      const { config } = new Config(overrides)

      expect(config).toStrictEqual({ ...defaultConfig, ...overrides })
      expect(config).toMatchSnapshot()
    })
  })
})
