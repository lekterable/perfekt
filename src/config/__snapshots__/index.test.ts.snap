// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config get config should return config with overrides 1`] = `
{
  "breakingHeader": "# BREAKING",
  "groups": [
    {
      "change": "minor",
      "name": "## Features",
      "types": [
        "feat",
        "feature",
      ],
    },
    {
      "change": "patch",
      "name": "## Fixes",
      "types": [
        "fix",
      ],
    },
  ],
  "ignoredScopes": [
    "changelog",
  ],
  "lineFormat": "* %message% %hash% %hash%",
  "miscHeader": "## Misc",
  "releaseHeader": "## Release version - @%version%",
  "unreleasedHeader": "## **Unreleased**",
}
`;

exports[`Config get config should return default config 1`] = `
{
  "breakingHeader": "## BREAKING",
  "groups": [
    {
      "change": "minor",
      "name": "## Features",
      "types": [
        "feat",
        "feature",
      ],
    },
    {
      "change": "patch",
      "name": "## Fixes",
      "types": [
        "fix",
      ],
    },
  ],
  "ignoredScopes": [
    "changelog",
  ],
  "lineFormat": "- %message% %hash%",
  "miscHeader": "## Misc",
  "releaseHeader": "# %version%",
  "unreleasedHeader": "# Latest",
}
`;
